modules/account-lib/src/coin/sol/transferBuilder.ts

```diff
@@ -1,37 +1,67 @@
 import { BaseCoin as CoinConfig } from '@bitgo/statics';
-import { BaseKey } from '../baseCoin/iface';
-import { NotImplementedError } from '../baseCoin/errors';
+import { BuildTransactionError } from '../baseCoin/errors';
 import { TransactionBuilder } from './transactionBuilder';
 import { Transaction } from './transaction';
-import { Transaction as SolTransaction } from '@solana/web3.js';
+import { isValidAmount, isValidPublicKey } from './utils';
 import { TransactionType } from '../baseCoin';
+import { InstructionBuilderTypes } from './constants';
+import { Memo, Transfer } from './iface';
 
 export class TransferBuilder extends TransactionBuilder {
   constructor(_coinConfig: Readonly<CoinConfig>) {
     super(_coinConfig);
   }
 
   protected get transactionType(): TransactionType {
-    return TransactionType.WalletInitialization;
+    return TransactionType.Send;
   }
 
-  /** @inheritdoc */
-  protected buildSolTransaction(): SolTransaction {
-    throw new NotImplementedError('buildImplementation not implemented');
+  /**
+   *  Set the memo
+   *
+   * @param {string} memo
+   * @returns {TransactionBuilder} This transaction builder
+   */
+  memo(memo: string): this {
+    this.validateMemo(memo);
+    const memoData: Memo = {
+      type: InstructionBuilderTypes.Memo,
+      params: { memo },
+    };
+    this._instructionsData.push(memoData);
+    return this;
   }
```

Comment:
this should be moved to the base transaction builder - memo's are supported for all transactions (and not just transfers)

Reply:
The thing is, do we want to support memos everywhere or just in transfers?

Reply:
memos should be supported everywhere.